annotator/src/main/java/org/cbioportal/annotator/internal/GenomeNexusImpl.java

```diff
@@ -532,4 +534,35 @@ private TranscriptConsequenceSummary getCanonicalTranscript(VariantAnnotation gn
         hgvspNullClassifications.add("RNA");
         return hgvspNullClassifications;
     }
+
+    @Override
+    public List<AnnotatedRecord> getAnnotatedRecordsUsingPOST(AnnotationSummaryStatistics summaryStatistics, List<MutationRecord> mutationRecords, String isoformOverridesSource, Boolean replace) throws Exception {
+        // construct list of genomic location objects to pass to api client
+        List<GenomicLocation> genomicLocations = new ArrayList<>();
```

Comment:
Since we are creating a new list of `GenomicLocations` to pass to the annotator, really this method could just take the original (unpartitioned) list of `MutationRecords` and a start and end index as parameters.  Your way is probably clearer, but is creating a lot of unnecessary lists.  Probably clarity of code is more important than any tiny gain in memory/efficiency.  Yeah I like your way best.